src/ck-core/ckfutures.C

```diff
@@ -426,6 +426,23 @@ public:
   }
 };
 
+// TODO :
+// a poorly located helper function for getting a buffer from a message
+// should be moved elsewhere (when PR#3228 is merged), and probably named ck::get_message_buffer(void*)
+void* CkGetMsgBuffer(void* msg) {
```

Comment:
Do you have any ideas for how this function could be more robust? Ideally it should work with any `message` defined in a .ci file.

Reply:
Ah, excellent. Glad there's already a potential use-case in mind, I hope for it to be useful (especially in conjunction with wait all/any).

Hmm, good question. Perhaps we can discuss in core later. Handling user-defined message types is tricky. In my mind, the simplest path would be to make a specializable template class (a la `std::hash`) that users can specialize for their own message types. Since a user's message's internal structure can vary widely, I think it would be best for them to statically provide the offset somehow -- as we cannot virtualize messages (AFAIK).

Reply:
Having thought about this for a bit and drawing a blank wrt a better solution, we may consider generating a list of "essential" internal message types and restrict `ck::future` to receiving messages of such types. Perhaps adding a disclaimer along the lines of:

"_In most cases, ``ck::future<T>`` cannot receive messages/values via ``get`` that have been not been sent/set by a corresponding call to ``set``. However, it can receive messages of some internal message types when sent via a call to `CkSendFuture`. Note, others may be wrapped as a PUP-able value and explicitly received as the expected type(s); for example, one might wrap a message as `CkMarshalledMsg`, sent via a `CkMarshallMsg`, then type-casted to the received message on the receiver-side_."

Reply:
I updated the code to support array-creation messages and docs, can you check to see if these updates are satisfactory?